common/src/main/scala/uk/ac/wellcome/models/Identifiable.scala

```diff
@@ -0,0 +1,11 @@
+package uk.ac.wellcome.models
+
+case class UnidentifiableException()
+    extends Exception("Can't find canonicalId for this Identifiable")
+
+trait Identifiable {
+  val canonicalId: Option[String]
+  def id: String = canonicalId.getOrElse(
+    throw UnidentifiableException()
```

Comment:
Worth a test that we really do throw this exception if you try to index

* A work without an ID?
* A work with an ID with an item without?

(Or have I missed those tests?)